### DISCUSSION TITLE
Query String and Table Filters

### DISCUSSION THREAD

**Discussion by MACscr** [filamentphp/filament#1088] [Ideas]
I figured I would open this discussion topic where we could brainstorm ideas for solving the inability to have a queryString for Table Filters. They are used pretty heavily within my app and hitting the back button to go back to a table after editing a record only to find that your filters are no longer applied is a bit frustrating. I know the issue is that filters are JSON based and livewire has no way to use those for the query strings. Has anyone besides Dan looked into solving this? Any ideas?

**Comment by ryangjchandler** [comment]
@MACscr Have you tried hooking up the query string yourself? What sort of scenarios don't work?

**Reply by MACscr** [reply]
@ryangjchandler all filters dont work with querystring because of how Dan has them setup as mentioned above. His words, not mine. Not blaming or anything, its just a current limitation of the table builder and livewire and why I brought it to discussion here so maybe the community can help figure something out as I know I am not the only one with the need.

**Reply by ryangjchandler** [reply]
I've been able to use filters with the query string, it's kind of integral to one of our tables at work.

**Reply by MACscr** [reply]
> I've been able to use filters with the query string, it's kind of integral to one of our tables at work.

Care to share how you are doing this? Seems kind of odd that @danharrin would say its not possible if you have been doing it all along.

**Reply by ryangjchandler** [reply]
You can do:

```
protected $queryString = ['tableFilters'];
```

On your Livewire component.

**Reply by MACscr** [reply]
Wow, got it working as you suggested in Discord by moving it from my resource to my ListRecords component:

```php
<?php

namespace App\Filament\Resources\UserResource\Pages;

use App\Filament\Resources\UserResource;
use Filament\Resources\Pages\ListRecords;

class ListUsers extends ListRecords
{
    public static string $resource = UserResource::class;

    protected $queryString = [
        'tableSortColumn',
        'tableSortDirection',
        'tableSearchQuery' => ['except' => ''],
        'tableFilters',
    ];
}
```

**Comment by faizananwerali** [comment]
how to disable query string. I hate it in my particular case.

**Reply by pboivin** [reply]
I think this should do : `protected $queryString = [];`

**Reply by faizananwerali** [reply]
Thanks. It worked. This should also be in the documentation. Spending hours trying to figure it out without breaking the code is a nightmare. Got my head scratching as to why there is a query string every time I'm trying to filter something. Thanks again.

**Reply by pboivin** [reply]
This is more of a Livewire thing, but I can see the benefit of having this in the Filament docs also. Feel free to submit a pull-request :)
